src/hooks/useNetworkMotionStatesAllColonies.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  getExtensionHash,
+  type MotionState,
+  VotingReputationFactory,
+} from '@colony/colony-js';
+import { Extension } from '@colony/colony-js';
+import { type Provider } from '@ethersproject/providers';
+import { useEffect, useState } from 'react';
+
+import { supportedExtensionsConfig } from '~constants/index.ts';
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useGetCurrentExtensionsVersionsQuery } from '~gql';
+import { notNull, notUndefined } from '~utils/arrays/index.ts';
+
+import { mapToInstalledExtensionData } from '../utils/extensions.ts';
+
+import useJoinedColoniesWithExtensions from './useJoinedColoniesWithExtensions.ts';
+
+export type MotionStatesMap = Map<string, MotionState | null>;
+
+export type MotionStatesMapByColonies = Map<string, MotionStatesMap>;
+
+export type RefetchMotionStates = (motionIdsToRefetch?: string[]) => void;
+
+const getInstalledExtensions = (colony, versionsData) => {
+  const colonyExtensions = colony?.extensions?.items?.filter(notNull);
+  if (!colonyExtensions) {
+    return [];
+  }
+
+  const extensionVersions =
+    versionsData?.listCurrentVersions?.items?.filter(notNull);
+
+  const colonyExtensionsMap = colonyExtensions.map((extension) => {
+    const extensionConfig = supportedExtensionsConfig.find(
+      (e) => getExtensionHash(e.extensionId) === extension?.hash,
+    );
+
+    const { version } =
+      extensionVersions?.find((e) => e?.extensionHash === extension.hash) || {};
+
+    // Unsupported extension
+    if (!extensionConfig || !version) {
+      return null;
+    }
+
+    return mapToInstalledExtensionData({
+      colony,
+      extensionConfig,
+      colonyExtension: extension,
+      version,
+    });
+  });
+  return colonyExtensionsMap.filter(notNull);
+};
+
+const getVotingReputationAdress = (installedExtensions): string | undefined => {
+  const votingReputationExtension = installedExtensions.find(
+    (extension) => extension.extensionId === Extension.VotingReputation,
+  );
+
+  return votingReputationExtension?.address;
+};
```

Comment:
Is all this processing necessary if you only care about the hash and the address of each colony extensions?